src/conf.c

```diff
@@ -135,8 +135,10 @@ static int validate_common(cfg_t *cfg, const char *provider, int custom)
 	}
 
 	if (!custom && !cfg_getstr(cfg, "username")) {
-		cfg_error(cfg, "Missing username setting for DDNS provider %s", provider);
-		return -1;
+		if (strcmp(provider, "default@freemyip.com") != 0) {	
```

Comment:
Just added a `.nousername = 1` flag that can be declared per provider in the plugins instead.  Hope I didn't break your code now.